### ISSUE TITLE
Read-only doesn't work for notebooks

### ISSUE THREAD

**Issue by alexr00** [microsoft/vscode#185320]
1. Have an extension that provides a read-only filesystem.
2. Open a notebook in that filesystem
3. Try to edit the file. The editor will let you edit.

Example with GitHub Pull Requests and Issues:
1. Create a PR that had notebook file in it.
2. Check out some other branch so that you're not on the PR branch anymore.
3. Open the Pull Requests view and expand the PR you just created
4. Open the notebook file from the PR
5. Try to edit the file. The editor will let you edit, even though the file system provider states that it's read-only.

**Comment by bpasero** [comment]
//cc @amunger

**Comment by bpasero** [comment]
I think this needs more support in `notebookEditorWidget.setOptions` to pass in cell editor options from the monaco editor:

https://github.com/microsoft/vscode/blob/70d6aa9e2dd427f8df0c631dc4545062ea6c2be0/src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts#L1179

We could then easily pass in the relevant readonly message down to the cell editor.

**Comment by rebornix** [comment]
`NotebookEditorWidget` already supports `readonly`, you can try by running command "Set Active File readonly in active session" and then you won't be able to edit cell content or insert/delete cells.

If files from readonly fs are editable, it means we are not getting the `readonly` state properly from the editor service.

**Comment by rebornix** [comment]
I can actually reproduce this even for text files on macOS

* Make a file readonly by context menu -> get info -> sharing & permission -> readonly
* Open the file
* Edit 🐛 , you can modify the file
* Save
<img width="462" alt="image" src="https://github.com/microsoft/vscode/assets/876920/b4f9c7de-dc71-45f5-8a72-c08da035664c">

From the save error, it seems the FS provider knows about the readonlyness but the monaco editor was not set to `readonly`.

**Comment by bpasero** [comment]
@rebornix for https://github.com/microsoft/vscode/issues/185320#issuecomment-1751474262 to work you need to configure `files.readonlyFromPermissions: true`. I do not think it is related to what AlexR is reporting here.

**Comment by bpasero** [comment]
@alexr00 I seem to actually not be able to reproduce this with GitHub issues notebook at least. Here is what I do:
* open https://github.com/microsoft/vscode-extension-samples/tree/main/fsprovider-sample
* mark the `memfs` as `isReadonly: true` for the call to `registerFileSystemProvider`
* let it create a `memfs:/file.github-issues` file with some typical notebook content
* run it and open `file.github-issues`
* the notebook is readonly

<img width="656" alt="image" src="https://github.com/microsoft/vscode/assets/900690/9f9f8940-f3c8-4d91-84c1-5eeb6c6b505d">

**Comment by alexr00** [comment]
I can still reproduce. Maybe it is some interaction with the diff editor?

1. Install GHPRI
2. Add the following to the setting `githubPullRequests.queries`:
```
	"githubPullRequests.queries": [
...
		{
			"label": "Readonly notebook bug",
			"query": "is:closed author:mjbvz Update notebook milestones"
		},
...
```
3. Open the Pull Requests view
4. Expand the "Readonly notebook bug" query and expand one of the PRs there
5. Open one of the notebook diff
6. ✅ the editor tab will show readonly
7. 🐛 the editor itself can still be edited

![Recording 2023-10-09 at 12 26 55](https://github.com/microsoft/vscode/assets/38270282/b41a6b83-6c42-4ebd-8485-91ae24689925)

**Comment by bpasero** [comment]
Yup, I can see how the diff editor for notebooks has editable cells.

@rebornix back to notebook land

**Comment by DonJayamanne** [comment]
Fixed
